toolkit/offchain/src/reserve/update_settings.rs

```diff
@@ -82,148 +192,23 @@ fn update_reserve_settings_tx(
-			&(&parameters).into(),
-			&reserve,
-			&test_governance_data(),
-			governance_script_cost(),
-			redeemer_script_cost(),
-			&test_transaction_context(),
-		)
-		.unwrap();
-
-		let body = tx.body();
-		let inputs = body.inputs();
-		let outputs = body.outputs();
-
-		// Both inputs are used to cover transaction
-		assert_eq!(
-			inputs.get(0).to_string(),
-			"1212121212121212121212121212121212121212121212121212121212121212#0"
-		);
-		assert_eq!(
-			inputs.get(1).to_string(),
-			"7474747474747474747474747474747474747474747474747474747474747474#3"
-		);
-		assert_eq!(
-			outputs.get(0).address(),
-			test_validator_script().address(cardano_serialization_lib::NetworkIdKind::Testnet)
-		);
-		assert_eq!(
-			outputs.get(1).address().to_hex(),
-			"6032230eeaaae0ff7a97e7f088e65874bd79c0fe2a99399f795e84543a"
-		);
-		assert_eq!(outputs.get(0).plutus_data().unwrap(), ReserveDatum::from(&parameters).into());
-	}
 
-	fn test_transaction_context() -> TransactionContext {
-		TransactionContext {
-			payment_key: payment_key(),
-			payment_key_utxos: vec![make_utxo(116u8, 3, 996272387, &payment_addr())],
-			network: NetworkIdKind::Testnet,
-			protocol_parameters: protocol_parameters(),
-		}
-	}
-
-	fn test_validator_script() -> PlutusScript {
-		PlutusScript { bytes: hex!("445566").to_vec(), language: Language::new_plutus_v2() }
-	}
-
-	fn test_auth_policy_script() -> PlutusScript {
-		PlutusScript { bytes: hex!("556677").to_vec(), language: Language::new_plutus_v2() }
-	}
-
-	fn test_ics_validator_script() -> PlutusScript {
-		PlutusScript { bytes: hex!("667788").to_vec(), language: Language::new_plutus_v2() }
-	}
-
-	fn test_governance_script() -> PlutusScript {
-		PlutusScript { bytes: hex!("112233").to_vec(), language: Language::new_plutus_v2() }
-	}
-
-	fn test_governance_input() -> OgmiosUtxo {
-		OgmiosUtxo { transaction: OgmiosTx { id: [16; 32] }, index: 0, ..Default::default() }
-	}
-
-	fn test_auth_policy_version_utxo() -> OgmiosUtxo {
-		OgmiosUtxo { transaction: OgmiosTx { id: [17; 32] }, index: 0, ..Default::default() }
-	}
-
-	fn test_validator_version_utxo() -> OgmiosUtxo {
-		OgmiosUtxo { transaction: OgmiosTx { id: [18; 32] }, index: 0, ..Default::default() }
-	}
-
-	fn test_ics_validator_version_utxo() -> OgmiosUtxo {
-		OgmiosUtxo { transaction: OgmiosTx { id: [19; 32] }, index: 0, ..Default::default() }
-	}
-
-	fn test_governance_data() -> GovernanceData {
-		GovernanceData { policy_script: test_governance_script(), utxo: test_governance_input() }
-	}
+	tx_builder.add_script_reference_input(
+		&reserve.auth_policy_version_utxo.to_csl_tx_input(),
+		reserve.scripts.auth_policy.bytes.len(),
+	);
+	tx_builder.add_script_reference_input(
+		&reserve.validator_version_utxo.to_csl_tx_input(),
+		reserve.scripts.validator.bytes.len(),
+	);
 
-	fn governance_script_cost() -> ExUnits {
-		ExUnits::new(&100u64.into(), &200u64.into())
-	}
+	tx_builder.add_required_signer(&ctx.payment_key_hash());
```

Comment:
Cosmetic: function below does it, hence this line is redundant. I have a PR open that fixes it in other places